### PR TITLE
Update to new services

### LINKED ISSUE #1088: Version conficts in jupyterlab and jupyter-js-services
@jasongrout and I were updating the jupyterlab_vega extension today and we found that the version of jupyter-js-service used by the latest release of jupyter (0.5.0) is behind the latest stable of jupyter-js-services itself. To build my extension I had to depend on jupyter-js-services, but that ended up conflicting with the version in jupyterlab. We had to monkey patch around all of this.

Can we release a minor version of jupyterlab that depends on latest jupyter-js-services?

### PR DESCRIPTION
**Pull request by blink1073**
This depends on https://github.com/jupyterlab/services/pull/249/files and a release of `@jupyterlab/services`.

Fixes #1088.

Uses the TS 2.0 typings handling where possible, with extra typings still needed for TypeDoc.

### CODE REVIEW

**Comment by vidartf** [comment]
Note that according to the [handbook](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#types-typeroots-and-types):
 > If types is specified, only packages listed will be included. For instance:
```
{
   "compilerOptions": {
       "types" : ["node", "lodash", "express"]
   }
}
```
This tsconfig.json file will only include ./node_modules/@types/node, ./node_modules/@types/lodash and ./node_modules/@types/express. Other packages under node_modules/@types/* will not be included.

Not sure if the actual behavior is completely after specs though.

**Comment by jasongrout** [comment]
Important to also read a paragraph or two down:

> Keep in mind that automatic inclusion is only important if you’re using files with global declarations (as opposed to files declared as modules). If you use an import "foo" statement, for instance, TypeScript may still look through node_modules & node_modules/@types folders to find the foo package.

**Comment by vidartf** [comment]
@jasongrout Thanks for clearing up my confusion :)

**Comment by blink1073** [comment]
This is finished, and I tested all of the examples.

File: examples/terminal/package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "watch": "watch \"npm run update && npm run build\" ../../src src --wait 10"
   },
   "dependencies": {
+    "@jupyterlab/services": "^0.22.0",
     "jupyter-js-services": "^0.21.0",
```
Comment (jasongrout) [inline examples/terminal/package.json]: jupyter-js-services is still in the dependency list here...

**Comment by jasongrout** [comment]
There is still a mention in tutorial directory:

```
% ack jupyter-js-services
tutorial/documents.md
83:to the session and contents API from jupyter-js-services for the

tutorial/plugins.md
36:- [Services](http://jupyterlab.github.io/jupyterlab/modules/_services_plugin_.html#servicesprovider) - An application-specific interface to `jupyter-js-services`.
```

**Comment by jasongrout** [comment]
Looks great; thanks!

**Pull request merged**